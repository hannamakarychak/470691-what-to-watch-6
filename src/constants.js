export const ALL_GENRES = `All genres`;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};
