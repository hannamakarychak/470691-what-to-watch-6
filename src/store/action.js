export const ActionType = {
  SET_GENRE: `main/setGenre`,
  GET_MOVIES_BY_GENRE: `main/getMoviesByGenre`
};

export const ActionCreator = {
  setGenre: (genre) => ({
    type: ActionType.SET_GENRE,
    payload: genre
  }),

  getListByGenre: () => ({
    type: ActionType.GET_MOVIES_BY_GENRE,
    payload: []
  })
};
