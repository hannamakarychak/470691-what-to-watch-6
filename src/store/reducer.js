import {ALL_GENRES, AuthorizationStatus} from "../constants";
import {ActionType} from "./action";

const initialState = {
  genre: ALL_GENRES,
  list: [],
  isDataLoaded: false,
  authorizationStatus: AuthorizationStatus.NO_AUTH
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case ActionType.GET_ALL_MOVIES:
      return {
        ...state,
        isDataLoaded: true,
        list: action.payload
      };

    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload
      };

    default:
      return state;
  }
};
