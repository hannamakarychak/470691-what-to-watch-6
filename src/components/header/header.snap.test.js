import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';

import Header from './header';
import {NameSpace} from '../../store/reducer';
import {AuthorizationStatus} from '../../constants';

const mockStore = configureStore({});


describe(`Should Header render correctly`, () => {
  it(`when isLoggedIn is true`, () => {
    const history = createMemoryHistory();
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        email: `email@test.com`,
        avatar: `https://avatar.com/face.png`,
        isLoaded: true
      },
    });
    const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <Header className="test-class">
            this is children
          </Header>
        </Router>
      </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it(`when isLoggedIn is false`, () => {
    const history = createMemoryHistory();
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        email: null,
        avatar: null,
        isLoaded: true
      },
    });
    const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <Header className="test-class">
            this is children
          </Header>
        </Router>
      </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
  it(`when isLoggedIn is false on /login route`, () => {
    const history = createMemoryHistory();
    history.push(`/login`);
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        email: null,
        avatar: null,
        isLoaded: true
      },
    });
    const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <Header className="test-class">
            this is children
          </Header>
        </Router>
      </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

