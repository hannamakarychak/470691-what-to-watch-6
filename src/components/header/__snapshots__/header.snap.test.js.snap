// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly when isLoggedIn is false 1`] = `
<div>
  <header
    class="page-header test-class"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    this is children
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
</div>
`;

exports[`Should Header render correctly when isLoggedIn is false on /login route 1`] = `
<div>
  <header
    class="page-header test-class"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    this is children
    <div
      class="user-block"
    />
  </header>
</div>
`;

exports[`Should Header render correctly when isLoggedIn is true 1`] = `
<div>
  <header
    class="page-header test-class"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    this is children
    <div
      class="user-block"
    >
      email@test.com
      <a
        class="user-block__avatar"
        href="/mylist"
      >
        <img
          alt="User avatar"
          height="63"
          src="https://avatar.com/face.png"
          width="63"
        />
      </a>
    </div>
  </header>
</div>
`;
